packages/synthetic-chain/src/lib/core-eval.ts

```diff
@@ -0,0 +1,245 @@
+import { execFileSync } from 'child_process'; // TODO: use execa
+import * as fsp from 'fs/promises';
+import * as path from 'path';
+import * as process from 'process';
+
+import { ZipReader } from '@endo/zip';
+import dbOpen from 'better-sqlite3';
+
+import assert from 'assert';
+import { makeAgd } from '@agoric/synthetic-chain/src/lib/agd-lib.js';
+import { agoric } from '@agoric/synthetic-chain/src/lib/cliHelper.js';
+import { voteLatestProposalAndWait } from '@agoric/synthetic-chain/src/lib/commonUpgradeHelpers.js';
+import { dbTool } from '@agoric/synthetic-chain/src/lib/vat-status.js';
+import { type WebCache } from '@agoric/synthetic-chain/src/lib/webAsset.js';
+import {
+  bundleDetail,
+  ensureISTForInstall,
+  flags,
+  getContractInfo,
+  loadedBundleIds,
+  txAbbr,
+} from '@agoric/synthetic-chain/src/lib/core-eval-support.js';
+import { step } from '@agoric/synthetic-chain/src/lib/logging.js';
+
+const makeFakeWebCache = (base: string): WebCache => {
+  return {
+    getText(segment: string) {
+      return fsp.readFile(path.join(base, segment), 'utf8');
+    },
+    async storedPath(segment: string) {
+      return path.join(base, segment);
+    },
+    async size(segment: string) {
+      const info = await fsp.stat(path.join(base, segment));
+      return info.size;
+    },
+    toString() {
+      return 'fake web cache';
+    },
+    async remove() {
+      console.warn('noop remove');
+    },
+  };
+};
+
+/**
+ * Provide access to the outside world via context.
+ *
+ * TODO: refactor overlap with mn2-start.test.js
+ *
+ * @param {*} t
+ * @param {object} io
+ */
+const makeTestContext = async staticConfig => {
```

Comment:
I'm confused by obvious test setup in a `/src/` directory. I'd guess this is because synthetic-chain is all for test support, but would it make sense to have everything in this package under `/test/` instead?

Reply:
this is legacy from when we used Ava to execute the steps. The next improvement would be to remove "test" terminology from this file. I'll take your comment as a request for that.